src/pages/level/Level.tsx

```diff
@@ -9,69 +9,258 @@ import BlocklyWorkspace, {
   type BlocklyWorkspaceProps,
   type ToolboxItemInfo,
 } from "./BlocklyWorkspace"
+import {
+  type ThreePanelLayout,
+  ThreePanelLayouts,
+  type TwoPanelLayout,
+  TwoPanelLayouts,
+  setThreePanelLayout,
+  setTwoPanelLayout,
+} from "../../app/slices"
+import { useAppDispatch, useSettings } from "../../app/hooks"
 import Controls from "./Controls"
 import { paths } from "../../routes"
-import { useSettings } from "../../app/hooks"
 
 export interface LevelProps {}
 
+type Direction = "horizontal" | "vertical"
 interface LevelState {
   panels: 2 | 3
   toolbox_contents: ToolboxItemInfo[]
 }
 
+const THREE_PANEL_VERTICAL_WITH_LEFT_HORIZONTAL = ThreePanelLayouts[1]
+
+/**
+ * Function that calculates the required panel configuration (default sizes,
+ * panel group directions, order of panels) based on the number of panels
+ * and preferred layout option
+ * Layout of panels contains 2 groups: inner and outer. Outer group is
+ * used only for "vertical split with horizontal left split" 3-panel
+ * layout. Inner group contains all 2 or 3 panels in all cases, except
+ * this one when the 3rd panel moves to the outer group, while the rest stay
+ * in the inner.
+ */
+function calculatePanelsConfig(
+  panels: number,
+  layout: TwoPanelLayout | ThreePanelLayout,
+): {
+  innerPGDirection: Direction
+  outerPGDirection: Direction
+  panels: {
+    /*
+      Array of default sizes of panels in %. First element is
+      the size of the panel containing the entire inner
+      panel group, followed by 2 or 3 numbers for panels
+      inside the inner panel group
+    */
+    defaultSizes: number[]
+    reverseOrder: boolean
+  }
+} {
+  let panelsDefaultSizes: number[],
+    reverseOrder = false,
+    innerPGDirection: Direction,
+    outerPGDirection: Direction = "horizontal"
+
+  if (panels === 2) {
+    panelsDefaultSizes = [100, 50, 50]
+    switch (layout) {
+      case "horizontal":
+        innerPGDirection = "vertical"
+        reverseOrder = true
+        break
+      case "auto":
+      case "vertical":
+      default:
+        innerPGDirection = "horizontal"
+        break
+    }
+  } else {
+    // 3 panels
+    switch (layout) {
+      case "verticalWithLeftHorizontal":
+        innerPGDirection = "vertical"
+        outerPGDirection = "horizontal"
+        panelsDefaultSizes = [50, 50, 50, 50]
+        break
+      case "horizontal":
+        innerPGDirection = "vertical"
+        panelsDefaultSizes = [100, 34, 33, 33]
+        reverseOrder = true
+        break
+      case "auto":
+      case "vertical":
+      default:
+        innerPGDirection = "horizontal"
+        panelsDefaultSizes = [100, 34, 33, 33]
+    }
+  }
+  return {
+    innerPGDirection,
+    outerPGDirection,
+    panels: {
+      defaultSizes: panelsDefaultSizes,
+      reverseOrder,
+    },
+  }
+}
+
+const Panel3Content: FC = () => (
```

Comment:
In the future, this 3rd component will contain a python code editor. Please rename this component and move it to it's own file at /workspace/frontend/rapid-router/src/pages/level/PythonEditor.tsx. You can then import this.

```suggestion
const PythonEditor: FC = () => (
```